File: 07_StacksAndQueues_Nesting.rb
# you can write to stdout for debugging purposes, e.g.
# puts "this is a debug message"

def solution(s)
  # write your code in Ruby 2.2
  cnt = 0 
  s.chars.each do |c|
    if c == "("
      cnt += 1
    else 
      return 0 if cnt == 0
      cnt -= 1 
    end
  end
  return cnt == 0 ? 1 : 0
end

File: Python/07_StacksAndQueues_Nesting.py
# you can write to stdout for debugging purposes, e.g.
# print("this is a debug message")

def solution(S):
    # write your code in Python 3.6
    
    # Bracketsより簡単なのでおしまい
    # 1種類のカッコしかないのでスタックすらいらずに，カウントするだけ
    cnt = 0
    for c in S:
        if c == "(": 
            cnt += 1
        else:
            if cnt == 0: return 0 
            cnt -= 1
    # end of for 

    return 1 if cnt == 0 else 0
# end of func